### PR TITLE
fix: strip PR numbers from commit

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**
fixes https://github.com/googleapis/synthtool/issues/596

### CODE REVIEW

File: tests/test_git_source.py [hunk 1]
```diff
@@ -248,3 +248,12 @@ def test_enumerate_versions_with_functionally_duplicate_sources():
         )
         assert 1 == ev.call_count
         assert [[v1, v2]] == versions
+
+
+def test_strip_pr_number():
+    strip = autosynth.git_source._strip_pr_number
+    assert "fix bugs", strip("fix bugs (#244)")
```
Comment (chingor13) [inline tests/test_git_source.py:255]: these assertions look incorrect
```suggestion
    assert "fix bugs" == strip("fix bugs (#244)")
```

File: autosynth/git_source.py [hunk 2]
```diff
@@ -24,6 +24,25 @@
 from autosynth import git, executor
 
 
+def _strip_pr_number(commit_subject: str) -> str:
```
Comment (chingor13) [inline autosynth/git_source.py:27]: I don't see this being used anywhere except the test

File: tests/test_git_source.py (hunk 1 above)
Comment (SurferJeffAtGoogle) [reply tests/test_git_source.py]: Doh.  Thanks!

File: autosynth/git_source.py (hunk 2 above)
Comment (SurferJeffAtGoogle) [reply autosynth/git_source.py:27]: Fixed.

**Pull request merged**